java/semantickernel-api/src/main/java/com/microsoft/semantickernel/services/ParentAIServiceSelector.java

```diff
@@ -0,0 +1,41 @@
+package com.microsoft.semantickernel.services;
+
+import com.microsoft.semantickernel.AIService;
+import com.microsoft.semantickernel.orchestration.KernelFunction;
+import com.microsoft.semantickernel.orchestration.contextvariables.KernelArguments;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ParentAIServiceSelector implements AIServiceSelector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParentAIServiceSelector.class);
+
+    protected final Map<Class<? extends AIService>, AIService> services;
+
+    public ParentAIServiceSelector() {
```

Comment:
Just a nit, but ctor's in an abstract class should be protected, which reinforces the fact that you can't construct the base class.